integration_tests/default-config_test.go

```diff
@@ -156,4 +160,64 @@ var _ = When("create default backstage", func() {
 
 	})
 
+	It("creates runtime object using raw configuration then updates StatefulSet to replace some immutable fields", func() {
+		if !*testEnv.UseExistingCluster {
+			Skip("Real cluster required to assert actual deletion and replacement of resources")
+		}
+
+		rawStatefulSetYamlContent := readTestYamlFile("raw-statefulset.yaml")
+		dbConf := map[string]string{"db-statefulset.yaml": rawStatefulSetYamlContent}
+
+		dbRaw := generateConfigMap(ctx, k8sClient, "dbraw", ns, dbConf, nil, nil)
+
+		backstageName := createAndReconcileBackstage(ctx, ns, bsv1alpha1.BackstageSpec{
+			RawRuntimeConfig: &bsv1alpha1.RuntimeConfig{
+				LocalDbConfigName: dbRaw,
+			},
+		}, "")
+
+		Eventually(func(g Gomega) {
+			By("creating Deployment")
+			deploy := &appsv1.Deployment{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Namespace: ns, Name: model.DeploymentName(backstageName)}, deploy)
+			g.Expect(err).ShouldNot(HaveOccurred())
+
+			By("creating StatefulSet")
+			ss := &appsv1.StatefulSet{}
+			name := fmt.Sprintf("backstage-psql-%s", backstageName)
+			err = k8sClient.Get(ctx, types.NamespacedName{Namespace: ns, Name: name}, ss)
+			g.Expect(err).ShouldNot(HaveOccurred())
+			g.Expect(ss.Spec.Template.Spec.Containers).To(HaveLen(1))
+			g.Expect(ss.Spec.Template.Spec.Containers[0].Image).To(Equal("busybox"))
+			g.Expect(ss.Spec.ServiceName).To(Equal("backstage-psql-db-statefulset-hl"))
+		}, time.Minute, time.Second).Should(Succeed())
+
+		By("updating CR with new DB raw config")
+		dbConf = map[string]string{
+			"db-statefulset.yaml": strings.ReplaceAll(rawStatefulSetYamlContent, "backstage-psql-db-statefulset-hl", fmt.Sprintf("backstage-psql-%s-hl", backstageName)),
+		}
+		update := &bsv1alpha1.Backstage{}
+		err := k8sClient.Get(ctx, types.NamespacedName{Name: backstageName, Namespace: ns}, update)
+		Expect(err).To(Not(HaveOccurred()))
+		update.Spec.RawRuntimeConfig.LocalDbConfigName = generateConfigMap(ctx, k8sClient, "dbraw-new", ns, dbConf, nil, nil)
+		err = k8sClient.Update(ctx, update)
+		Expect(err).To(Not(HaveOccurred()))
+
+		// Patching StatefulSets is done by the reconciler in two passes: first deleting the StatefulSet first, then recreating it in the next reconcilation.
+		for i := 0; i < 2; i++ {
+			_, err = NewTestBackstageReconciler(ns).ReconcileAny(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{Name: backstageName, Namespace: ns},
+			})
+			Expect(err).To(Not(HaveOccurred()))
+		}
+
+		Eventually(func(g Gomega) {
+			By("replacing StatefulSet")
+			dbStatefulSet := &appsv1.StatefulSet{}
+			err = k8sClient.Get(ctx, types.NamespacedName{Namespace: ns, Name: fmt.Sprintf("backstage-psql-%s", backstageName)}, dbStatefulSet)
+			g.Expect(err).ShouldNot(HaveOccurred())
+			g.Expect(dbStatefulSet.Spec.ServiceName).To(Equal(fmt.Sprintf("backstage-psql-%s-hl", backstageName)))
```

Comment:
it will be backstage-psql-<backstageName> after https://github.com/janus-idp/operator/pull/371